File: apps/admin/src/types/vehicle.ts
export type VehicleType = 'sedan' | 'suv' | 'pickup' | 'van' | 'motorcycle' | 'other';

export interface Vehicle {
  id: string;
  owner_id: string;
  plate: string;
  brand: string | null;
  model: string | null;
  color: string | null;
  type: VehicleType;
  created_at: string;
  owner?: { name: string; email: string };
}

File: apps/mobile/lib/features/vehicles/domain/entities/vehicle.dart
class Vehicle {
  final String id;
  final String plate;
  final String? brand;
  final String? model;
  final String? color;
  final String type;
  final String? ownerName;

  const Vehicle({
    required this.id,
    required this.plate,
    this.brand,
    this.model,
    this.color,
    this.type = 'sedan',
    this.ownerName,
  });
}
